burn-core/src/grad_clipper/base.rs

```diff
@@ -0,0 +1,96 @@
+use crate::tensor::Tensor;
+use burn_tensor::{backend::Backend, ElementConversion};
+
+/// Gradient Clipper provides a way to mitigate exploding gradients
+/// by clipping every component of the gradient to a value during
+/// backpropagation.
+pub enum GradientClipper {
+    ClipByValue(f32),
+    ClipByNorm(f32),
+}
+
+impl GradientClipper {
+    pub fn clip_gradient<B: Backend, const D: usize>(&self, grad: Tensor<B, D>) -> Tensor<B, D> {
+        match self {
+            GradientClipper::ClipByValue(threshold) => self.clip_by_value(grad, *threshold),
+            GradientClipper::ClipByNorm(max_norm) => self.clip_by_norm(grad, *max_norm),
+        }
+    }
+
+    fn clip_by_value<B: Backend, const D: usize>(
+        &self,
+        grad: Tensor<B, D>,
+        threshold: f32,
+    ) -> Tensor<B, D> {
+        let mut grad_data = grad.to_data();
+        grad_data.value.iter_mut().for_each(|val| {
+            let f_val: f32 = val.elem();
+            if f_val > threshold {
+                *val = threshold.elem();
+            } else if f_val < -threshold {
+                *val = (-threshold).elem();
+            }
+        });
+
+        Tensor::from_data_device(grad_data, &grad.device())
+    }
+
+    fn l2_norm<B: Backend, const D: usize>(tensor: &Tensor<B, D>) -> Tensor<B, 1> {
```

Comment:
I would use a owned tensor here instead and call `clone()` in the `clip_by_norm` function.